python/lsst/ts/athexapod/controller.py

```diff
@@ -80,36 +73,30 @@ def is_connected(self):
         connected : `bool`
 
         """
-        return self.reader is not None and self.writer is not None
+        return self.client.connected
 
     async def connect(self):
         """Connect to hexapod controller."""
 
-        async with self.lock:
-            if self.is_connected:
-                raise RuntimeError(
-                    "Reader or Writer not None. Try disconnecting first."
-                )
-
-            connect_task = asyncio.open_connection(host=self.host, port=self.port)
-
-            self.reader, self.writer = await asyncio.wait_for(
-                connect_task, timeout=self.long_timeout
-            )
+        self.client = tcpip.Client(
+            host=self.host,
+            port=self.port,
```

Comment:
Where are `self.host` and `self.port` defined? They were removed from the `__init__` function.

Reply:
They are defined in the configure method but I added them back in the __init__ since that is confusing and prone to bugs.

Reply:
Setting them in the config method is not enough. All attributes need to be defined in the `__init__` method. Thanks for adding them back.